contrib/epee/src/wipeable_string.cpp

```diff
@@ -109,9 +115,112 @@ void wipeable_string::push_back(char c)
   buffer.back() = c;
 }
 
-void wipeable_string::pop_back()
+void wipeable_string::operator+=(char c)
+{
+  push_back(c);
+}
+
+void wipeable_string::operator+=(const std::string &s)
+{
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - s.size(), "Appended data too large");
+  grow(size(), size() + s.size());
+  const char *ptr = s.c_str();
+  while (*ptr)
+    push_back(*ptr++);
+}
+
+void wipeable_string::operator+=(const epee::wipeable_string &s)
+{
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - s.size(), "Appended data too large");
+  grow(size(), size() + s.size());
+  const char *ptr = s.data();
+  size_t sz = s.size();
+  while (sz--)
+    push_back(*ptr++);
+}
+
+void wipeable_string::operator+=(const char *s)
+{
+  size_t slen = strlen(s);
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - slen, "Appended data too large");
+  grow(size(), size() + slen);
+  while (*s)
+    push_back(*s++);
+}
+
+void wipeable_string::append(const char *data, size_t len)
+{
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - len, "Appended data too large");
+  grow(size(), size() + len);
+  while (len--)
+    push_back(*data++);
+}
+
+void wipeable_string::trim()
+{
+  size_t prefix = 0;
+  while (prefix < size() && data()[prefix] == ' ')
```

Comment:
I suppose technically it does matter since this is now a "generic" routine as it's in the class.
If a user were to enter funky space like chars somehow in seed input, then it'd possibly matter, though it'd be user punches himself then complains. I suppose you don't want me to use isspace from ctype.h, right ?

Reply:
Boost was doing that internally. So that would be no change from the existing implementation. Boost probably used the C++ locale, so that it was cached in case another thread changed the locale during processing.

Reply:
I assume by "the C++ locale", you don't mean something like "the C locale", right ? Because AFAIK that uses ASCII.

Otherwise, searching a bit, I see: std::ispace(c, locale("")); which seems to use the "current locale". Does that fit what you were asking for ?

Reply:
Boost grabs a C++ object representing the locale settings at time of creation. It is often the C locale, but not always. The C locale (the actual locale, not the current locale set in the C runtime) has many "implementation defined" things including most of the character encoding details. In (nearly?) all situations the C locale is using ASCII.

It occurred to me when doing this review, that my changes to pod->hex conversions changed from locale based to ASCII compatible only. No one complained - although I'm not sure what to make of that other than UTF8 probably dominates everything now.

This is a slightly different situation since input is involved - but making it work with ASCII compatible only systems probably won't blow everything up. So take the comment as "I made an observation, do with this information as you like".